Dockerfile

```diff
@@ -21,40 +21,21 @@ FROM returntocorp/ocaml:alpine-2022-06-09@sha256:99b453a838c9d94414991c0fd7be471
 USER root
 # for ocaml-pcre now used in semgrep-core
 # TODO: update root image to include python 3.9
-RUN apk add --no-cache pcre-dev python3 python3-dev &&\
-     pip install --no-cache-dir pipenv==2022.6.7
+RUN apk add --no-cache pcre-dev python3 python3-dev
+RUN pip install --no-cache-dir pipenv==2022.6.7
 
 USER user
 
 ENV OPAMYES=true
 
-WORKDIR /semgrep/semgrep-core/src/ocaml-tree-sitter-core
-COPY --chown=user semgrep-core/src/ocaml-tree-sitter-core/ .
-RUN ./configure \
- && ./scripts/install-tree-sitter-lib
-
-WORKDIR /semgrep/semgrep-core/src/pfff
-COPY --chown=user semgrep-core/src/pfff/*.opam .
-WORKDIR /semgrep/semgrep-core/src/ocaml-tree-sitter-core
-COPY --chown=user semgrep-core/src/ocaml-tree-sitter-core/*.opam .
-WORKDIR /semgrep/semgrep-core
-COPY --chown=user semgrep-core/*.opam .
-
-RUN opam install --deps-only \
-     /semgrep/semgrep-core/src/pfff \
-     /semgrep/semgrep-core/src/ocaml-tree-sitter-core \
-     /semgrep/semgrep-core
-
+# TODO: why copy the current dir to /semgrep and run the command from there?
+COPY --chown=user . /semgrep
```

Comment:
I'm not sure why there were different partial COPY instructions instead of just one. Normally, this is done to exclude some content. For this, it's best to rely on the `.dockerignore` mechanism, though. So, lacking an justification for the previous complexity, it's good that it's now simpler.

Reply:
Exactly. I could understand than for the final docker image (The python part), we might want to optimize and not copy everything, but anyway for the semgrep-core docker image we build in step1, we just care about the semgrep-core binary, so we don't care about the rest of the docker image.

Reply:
Sometimes, we want to exclude random junk to increase determinism, but I don't see how that would be a valid argument here.